src/dataflow/src/source/kafka.rs

```diff
@@ -180,29 +129,97 @@ impl KafkaSourceReader {
             &config_options,
         );
         let (stats_tx, stats_rx) = crossbeam_channel::unbounded();
-        let consumer: BaseConsumer<GlueConsumerContext> = kafka_config
-            .create_with_context(GlueConsumerContext {
-                activator: consumer_activator,
-                stats_tx,
-            })
-            .expect("Failed to create Kafka Consumer");
+        let consumer: Arc<BaseConsumer<GlueConsumerContext>> = Arc::new(
+            kafka_config
+                .create_with_context(GlueConsumerContext {
+                    activator: consumer_activator,
+                    stats_tx,
+                })
+                .expect("Failed to create Kafka Consumer"),
+        );
 
         let start_offsets = kc.start_offsets.iter().map(|(k, v)| (*k, v - 1)).collect();
 
+        let partition_info = Arc::new(Mutex::new(None));
+        {
+            let partition_info = Arc::clone(&partition_info);
+            let topic = topic.clone();
+            let consumer = Arc::clone(&consumer);
+            let metadata_refresh_frequency = config_options
+                .get("topic.metadata.refresh.interval.ms")
+                // Safe conversion: statement::extract_config enforces that option is a value
+                // between 0 and 3600000
+                .map(|s| Duration::from_millis(s.parse().unwrap()))
+                // Default value obtained from https://github.com/edenhill/librdkafka/blob/master/CONFIGURATION.md
+                .unwrap_or_else(|| Duration::from_millis(30000));
```

Comment:
The default for `topic.metadata.refresh.interval.ms` is `300000` fwiw (one more zero than above).

Reply:
thanks, I also changed it to `from_secs(300)`